### DISCUSSION TITLE
Can't register. Error code 403.

### DISCUSSION THREAD

**Discussion by ReiSuzunami** [walinejs/waline#1902] [Q&A] [answered]
Well I integrated Waline in Docker Compose with MongoDB on my own instances, not using any public services.

When I firstly configured the Docker Compose, I found that I had to add "?authSource=admin" after my Database's name or the comments could not even be sent. (Yes my MongoDB was in container too, as a dependency of waline and I operated it with the init root user in 'environment:')

Then, when I tried to use the register function of Waline, after filling out all the information, the system throwed an 403 error to me like below.

I checked the log of the container and it said: 
```
waline | mongodb://root:passwd@mongo:27017/comments?authSource=admin? 
waline | ForbiddenError: Forbidden
waline  |     at Object.throw (/app/node_modules/koa/lib/context.js:97:11)
waline  |     at module.exports.__before (/app/node_modules/@waline/vercel/src/logic/base.js:42:30)
waline  |     at /app/node_modules/think-logic/index.js:33:42
waline  |     at dispatch (/app/node_modules/koa-compose/index.js:42:32)
waline  |     at /app/node_modules/think-router-rest/index.js:31:12
waline  |     at dispatch (/app/node_modules/koa-compose/index.js:42:32)
waline  |     at Router.parseRule (/app/node_modules/think-router/router.js:266:17)
waline  |     at Router.run (/app/node_modules/think-router/router.js:291:19)
waline  |     at router (/app/node_modules/think-router/index.js:107:21)
waline  |     at dispatch (/app/node_modules/koa-compose/index.js:42:32)
```

I searched the Internet and did not found an answer. There was a similar thread #1338 but it was 404. Could someone help me?

![image](https://user-images.githubusercontent.com/116536096/227907139-d5abd463-3c90-45b8-abe8-d2d232278703.png)

**Comment by ReiSuzunami** [comment]
Well I found that there could not be two or more domains in SECURE_DOMAINS env. 
I wrote: "['example1.com','example2.com']" and "example1.com,example2.com" and neither of them worked.
finally I gave up using 2 domains and left only 1 there then everythings went right.

**Comment by lizheming** [comment, marked as answer]
403 Error is that accessing domain names are not in Secure_Domains. The format of Secure_DoMains is multiple domain names of the comma connection. If the configuration is not effective, it is recommended to check the configuration.
https://github.com/walinejs/waline/blob/main/packages/server/src/config/config.js#L115
https://github.com/walinejs/waline/blob/main/packages/server/src/logic/base.js#L23-L44
